src/gluonts/torch/model/forecast_generator.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Any, Iterator, List, Optional
+
+import numpy as np
+import torch
+from torch import nn
+
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.loader import DataLoader
+from gluonts.model.forecast_generator import ForecastGenerator, OutputTransform
+from gluonts.torch.model.forecast import DistributionForecast
+from gluonts.torch.modules.distribution_output import DistributionOutput
+
+
+def _extract_instances(x: Any) -> Any:
+    """
+    Helper function to extract individual instances from batched
+    mxnet results.
+    For a tensor `a`
+      _extract_instances(a) -> [a[0], a[1], ...]
+    For (nested) tuples of tensors `(a, (b, c))`
+      _extract_instances((a, (b, c)) -> [(a[0], (b[0], c[0])), (a[1], (b[1], c[1])), ...]
+    """
+    if isinstance(x, (np.ndarray, torch.Tensor)):
```

Comment:
This could be just factored out in one function which operates both on mxnet and torch